### PR TITLE
Move PySide builds to allowed failures

### PR DESCRIPTION
**Pull request by astrofrog**
Not quite sure why this is happening but it seems to be pretty systematic

### CODE REVIEW

**Comment by astrofrog** [comment]
Hmm some other PySide builds are running into issues now

**Comment by astrofrog** [comment]
@dhomeier - I'm moving all the PySide builds to allowed failures because they are failing non-deterministically with a segmentation fault. If you have any ideas what this could be due to, let me know! But for now I need to move these otherwise the failures interfere with the auto release workflow.

File: .github/workflows/ci_workflows.yml [hunk 1]
```diff
@@ -86,6 +83,12 @@ jobs:
           - enchant
           - hdf5
       envs: |
+
+        # Non-deterministic segmentation faults
```
Comment (dhomeier) [inline .github/workflows/ci_workflows.yml:87]:
```suggestion
        # Non-deterministic QThread exceptions
```
Those, and the macos-pyside63 one, are actually all failing on the
> QThread: Destroyed while thread is still running
test return, but that seems to be non-deterministic nonetheless (timing issue?)

**Review by dhomeier — APPROVED** [review]
Just trying to keep in mind one has to look for `QThread` calls, but the solution for the moment would be just the same. On macOS Ventura PySide 6.3 seems barely supported at all, segfaulting as soon as I try to start the tests. Would perhaps be more productive to look into the `layerTree` failure with PySide 6.4.

File: doc/conf.py [hunk 2]
```diff
@@ -139,7 +139,8 @@
                   ('py:class', 'glue.viewers.image.layer_artist.RGBImageLayerBase'),
                   ('py:class', 'glue.viewers.image.state.BaseImageLayerState'),
                   ('py:class', 'sip.voidptr'),
-                  ('py:class', 'PyQt5.sip.voidptr')]
+                  ('py:class', 'PyQt5.sip.voidptr'),
+                  ('py:class', 'PYQT_SLOT')]
```
Comment (dhomeier) [inline doc/conf.py:143]:
```suggestion
                  ('py:class', 'QKeySequence.StandardKey')],
                  ('py:class', 'PYQT_SLOT')]
```
Another one, in `BasicToolbar.grabShortcut`?

**Pull request merged**